tasks/upgradeContract.ts

```diff
@@ -0,0 +1,189 @@
+import { task } from "hardhat/config";
+import * as helpers from "../scripts/utils/helpers";
+import { getAddresses } from "../configs/addresses";
+import { _createSIP } from "./sips/createSIP";
+import { ISipArgument } from "./sips/args/SIPArgs";
+
+task("upgrade:massetManager", "Upgrade implementation of massetManager contract")
+.addOptionalParam("isMultisig", "flag if transaction needs to be intiated from the multisig contract")
+.addOptionalParam("isSIP", "flag if transaction needs to be initiated from the SIP")
+.setAction(async ({ isMultisig, isSIP }, hre) => {
+  helpers.injectHre(hre);
+  const { ethers, network, getNamedAccounts } = hre;
+  const { deployer } = await getNamedAccounts();
+  const configAddresses = getAddresses(network.name)
+  const myntAdminProxy = await ethers.getContract("MyntAdminProxy");
+  const massetManagerProxy = await ethers.getContract("MassetManager");
+
+  const MassetManagerFactory = await ethers.getContractFactory("MassetManager");
+  const newMassetManagerImpl = await MassetManagerFactory.deploy();
+  console.log(`Upgrading massetManager implementation to ${newMassetManagerImpl.address}`)
+
+  if(isMultisig) {
```

Comment:
it will not work if no `isMultisig` or `isSip` passed  
those are props that have clear network type dependency and are very unlikely to change . 
e.g. "mainnet" set for the mainnet network and for the forked mainnet will do the job.  
set "forked" tags to the mainnet and testnet for some specific functionality run on forked networks only (use timetravel, balance setting, impersonating etc.) 
you can keep the parameters `isMultisig` or `isSip` optional though to cover exceptions which should be a very rare case though

Reply:
Removed the unused conidtional